packages/react-compose/src/createClassResolver.test.ts

```diff
@@ -0,0 +1,47 @@
+import { createClassResolver } from './createClassResolver';
+
+describe('createClassResolver', () => {
+  const classResolver = createClassResolver({
+    root: 'root',
+    slot1: 'slot1',
+    slot2: 'slot2',
+    primary: 'primary',
+    size_small: 'small',
+    size_medium: 'medium',
+  });
+  const slots = { root: null, slot1: null, slot2: null };
+
+  it('can resolve slot classes', () => {
+    expect(classResolver({}, slots)).toEqual({
+      root: 'root',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve modifiers', () => {
+    expect(classResolver({ primary: true }, slots)).toEqual({
+      root: 'root primary',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve enums', () => {
+    // Can resolve
+    expect(classResolver({ size: 'small' }, slots)).toEqual({
+      root: 'root small',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve mixed content, including a className in props', () => {
```

Comment:
lets add test for when `state` has properties that are not mapped to classes:
```
classResolver({ className: 'foo', primary: true, size: 'medium', bar: 'bar' }, slots)
```

Reply:
updated.